### ISSUE TITLE
Default groups

### ISSUE THREAD

**Issue by ipspace** [ipspace/netlab#912]
It would be nice to have a way to define project-wise[^PG] groups. These groups (defined in **defaults.groups**) would be merged with the topology groups as follows:

* Trim members of default groups to nodes present in current topology
* For groups defined in topology, merge attributes. Existing group members would not be replaced or augmented, but if we have a group that does not have members, its members could be inherited from the defaults
* Copy default groups that are not defined in topology if they have at least one valid member
* Cleanup: remove groups with no members

The merge would have to happen before the lab topology group attributes are validated. Ideally we would:

* Validate default groups (to better pinpoint the source of validation errors)
* Merge default groups with active groups if we encountered no validation errors
* Validate lab topology groups

[^PG]: project = multiple topologies, potentially in different directories, see ipspace/bgplab for an example
